mne/event.py

```diff
@@ -1422,3 +1424,82 @@ def get_condition(self, raw, condition=None, stim_channel=None, mask=None,
             conds_data.append(dict(events=cat_t0, event_id=cat_id,
                                    tmin=tmin, tmax=tmax))
         return conds_data[0] if len(conds_data) == 1 else conds_data
+
+
+def match_event_names(event_names, keys, *, on_missing='raise'):
+    """Search a collection of event names for matching (sub-)groups of events.
+
+    This function is particularly helpful when using grouped event names
+    (i.e., event names containing forward slashes ``/``). Assuming for example
+    the following grouped event names in the data, you could easily query for
+    all ``auditory`` and ``left`` event names::
+
+        >>> event_names = [
+        ...     'auditory/left',
+        ...     'auditory/right',
+        ...     'visual/left',
+        ...     'visual/right'
+        ... ]
+        >>> match_event_names(
+        ...     event_names=event_names,
+        ...     keys=['auditory', 'left']
+        ... )
+        ['auditory/left', 'auditory/right', 'visual/left']
+
+    Parameters
+    ----------
+    event_names : array-like of str | dict
+        Either a collection of event names, or the ``event_id`` dictionary
+        mapping event names to event codes.
+    keys : array-like of str | str
+        One or multiple event names or groups to search for in ``event_names``.
+    on_missing : 'raise' | 'warn' | 'ignore'
+        How to handle situations when none of the ``keys`` can be found in
+        ``event_names``. If ``'warn'`` or ``'ignore'``, an empty list will be
+        returned.
+
+    Returns
+    -------
+    matches : list of str
+        All event names that match any of the ``keys`` provided.
+
+    Notes
+    -----
+    .. versionadded:: 1.0
+    """
+    _check_on_missing(on_missing)
+
+    if isinstance(event_names, dict):
+        event_names = list(event_names.keys())
```

Comment:
dict iteration is already over keys
```suggestion
        event_names = list(event_names)
```